src/lib/viewers/image/__tests__/ImageViewer-test.js

```diff
@@ -895,4 +895,26 @@ describe('lib/viewers/image/ImageViewer', () => {
             expect(image.cache.set).toBeCalledWith(IMAGE_FTUX_CURSOR_SEEN_KEY, true, true);
         });
     });
+
+    describe('isDiscoverabilityEnabled()', () => {
+        test.each`
+            canShow  | enableAnnotationsImageDiscoverability
+            ${true}  | ${true}
+            ${true}  | ${false}
+            ${false} | ${true}
+            ${false} | ${false}
+        `(
+            'should return correct value for isDiscoverabilityEnabled',
+            ({ canShow, enableAnnotationsImageDiscoverability }) => {
+                image.options.experiences = {
+                    tooltipFlowAnnotationsExperience: {
+                        canShow,
+                    },
+                };
+                image.options.enableAnnotationsImageDiscoverability = enableAnnotationsImageDiscoverability;
+
+                expect(image.isDiscoverabilityEnabled()).toBe(!canShow && !!enableAnnotationsImageDiscoverability);
```

Comment:
Instead of computing the expected result and embedding logic in the test, could it be passed in as a parameter?